latency.go

```diff
@@ -92,6 +92,9 @@ func main() {
 	// Make sure interest is set for subscribe before publish since a different connection.
 	c2.Flush()
 
+	// Allow time for subscription propagation when using clustered servers.
+	time.Sleep(1 * time.Second)
```

Comment:
This will work, but be better to make something that is deterministic.

Reply:
Will do..

Reply:
There are a few solutions that come to mind.

1) We can begin counting after the subscription has received a message by checking the `received` var via sync.atomic.  This has more impact to the codebase, but is deterministic.

2) Add logic to timeout the wait group.  While message rate is honored the reported test duration may be schewed.  This is simpler.  It'd replace `wg.Wait()` with this:
```go
	// There is a chance we missed some early messages if the route
	// wasn't setup.  This avoids a hang if we did not receive all
	// of the messages expected.
	c := make(chan struct{})
	go func() {
		defer close(c)
		wg.Wait()
	}()
	select {
	case <-c:
	case <-time.After(time.Second * 1):
		log.Printf("Warning:  Possible message loss.")
	}
```

I like the second option as it provides slack for longer running tests on flaky networks.

Do you have a preference?